src/Features/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -120,16 +120,6 @@ public ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities)
             // Using VS server capabilities because we have our own custom client.
             capabilities.OnAutoInsertProvider = new VSInternalDocumentOnAutoInsertOptions { TriggerCharacters = ["'", "/", "\n"] };
 
-            if (!supportsVsExtensions)
-            {
-                capabilities.DiagnosticOptions = new DiagnosticOptions
```

Comment:
this was the root cause of duplicate diagnostics in vscode - we were registering for requests with no identifier (and then also requests with an identifier in dynamic registration)